### PR TITLE
Custom Fields: Prepare list view and viewmodel to be reusable

### PR DESCRIPTION
**Pull request by hafizrahman**
Part of: #13493 

## Description

This PR mainly updates `OrderCustomFieldsDetails` and its related viewmodel to be `CustomFieldsDetails` by renaming them, moving the path out of Order, and generally making it to be usable by Products later on, too.

Additionally this also adds `EditableCustomFieldRow`, which is a new design for each custom field row, gated behind feature flag. It's intentionally similar to the original field, just limiting the value to three lines max, and adding a chevron. The design is still under discussion so it doesn't need too much attention for now, it just needs to be there for future uses.

## Steps to reproduce

0. Prepare an Order that contains some custom fields in it,
1. Run app,
2. Go to Orders,
3. Open the Order that has custom fields in it,
4. Tap the "View Custom Fields" button,
5. Make sure the fields are shown properly,
6. Make sure nothing looks glaringly wrong in the design:

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-ios/milestone/219">20.1</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/ViewRelated/Custom Fields/CustomFieldsDetails.swift [hunk 1]
```diff
@@ -1,19 +1,28 @@
 import SwiftUI
 
-struct OrderCustomFieldsDetails: View {
+struct CustomFieldsDetails: View {
```
Comment (itsmeichigo) [inline WooCommerce/Classes/ViewRelated/Custom Fields/CustomFieldsDetails.swift]: Nit: Please rename the view to clarify that it's a view. The postfix `View` is usually used for this.

File: WooCommerce/Classes/ViewRelated/Custom Fields/CustomFieldsDetails.swift [hunk 2]
```diff
@@ -1,19 +1,28 @@
...
     @Environment(\.presentationMode) var presentationMode
 
-    let customFields: [OrderCustomFieldsViewModel]
+    let viewEditCustomFieldsInProductsAndOrdersEnabled: Bool
```
Comment (itsmeichigo) [inline WooCommerce/Classes/ViewRelated/Custom Fields/CustomFieldsDetails.swift]: Super nit: the naming is a bit verbose making it hard to follow. How about `isEditable`?

File: WooCommerce/Classes/ViewRelated/Custom Fields/CustomFieldsDetails.swift [hunk 3]
```diff
@@ -90,13 +99,75 @@ private struct CustomFieldRow: View {
             Spacer()
         }
         .padding([.top, .bottom], Constants.hStackPadding)
+        .padding(.horizontal, Constants.hStackPadding)
```
Comment (itsmeichigo) [inline WooCommerce/Classes/ViewRelated/Custom Fields/CustomFieldsDetails.swift]: Nit: this can be combined into `.padding(Constants.hStackPadding)` to set the same padding for all edges.

File: WooCommerce/Classes/ViewRelated/Custom Fields/CustomFieldsDetails.swift [hunk 4]
```diff
@@ -90,13 +99,75 @@ private struct CustomFieldRow: View {
...
+
+            Spacer()
+
+            // Chevron icon
+            Image(uiImage: .chevronImage)
+                .flipsForRightToLeftLayoutDirection(true)
+                .foregroundStyle(Color(.textTertiary))
+        }
+        .padding([.top, .bottom], Constants.hStackPadding)
+        .padding(.horizontal, Constants.hStackPadding)
```
Comment (itsmeichigo) [inline WooCommerce/Classes/ViewRelated/Custom Fields/CustomFieldsDetails.swift]: Same comment as above about padding.

File: WooCommerce/Classes/ViewRelated/Custom Fields/CustomFieldsDetails.swift [hunk 5]
```diff
@@ -111,13 +182,25 @@ private extension CustomFieldRow {
...
 
+
+private extension EditableCustomFieldRow {
+    enum Constants {
+        static let spacing: CGFloat = 8
+        static let vStackPadding: CGFloat = 16
+        static let hStackPadding: CGFloat = 10
+        static let height: CGFloat = 64
+    }
+}
```
Comment (itsmeichigo) [inline WooCommerce/Classes/ViewRelated/Custom Fields/CustomFieldsDetails.swift]: Nit: instead of duplicating the constants, you can bring the enum to `CustomFieldsDetails` extension and reuse it for both `CustomFieldRow` and `EditableCustomFieldRow`.

**Review by itsmeichigo — COMMENTED** [review]
Since the UI is essentially the same for the custom field row except for the line limit and the chevron, can we just use the same view but add a bool to update the view as needed?

I also left a few suggestions in the comments.

File: WooCommerce/Classes/ViewRelated/Custom Fields/CustomFieldsDetails.swift (hunk 1 above)
Comment (hafizrahman) [reply WooCommerce/Classes/ViewRelated/Custom Fields/CustomFieldsDetails.swift]: Sure, updated in https://github.com/woocommerce/woocommerce-ios/pull/13639/commits/27ec9b5beb935f8f34f492ceb2e5ac5acfb6761b

File: WooCommerce/Classes/ViewRelated/Custom Fields/CustomFieldsDetails.swift (hunk 2 above)
Comment (hafizrahman) [reply WooCommerce/Classes/ViewRelated/Custom Fields/CustomFieldsDetails.swift]: Done in https://github.com/woocommerce/woocommerce-ios/pull/13639/commits/e23c6d26008ae66a7f7dd5818d604828245b62de

**Comment by hafizrahman** [comment]
> Since the UI is essentially the same for the custom field row except for the line limit and the chevron, can we just use the same view but add a bool to update the view as needed?

That makes sense. I was thinking that it's simpler to reason about when there's a separate view, instead of having to check for multiple bool checks, but I think in this case it's not too complicated. Added in https://github.com/woocommerce/woocommerce-ios/pull/13639/commits/1706a3fa8a3a6a4e9abe47892321f5971b050da2

File: WooCommerce/Classes/ViewRelated/Custom Fields/CustomFieldsDetails.swift (hunk 5 above)
Comment (hafizrahman) [reply WooCommerce/Classes/ViewRelated/Custom Fields/CustomFieldsDetails.swift]: Updated as part of https://github.com/woocommerce/woocommerce-ios/pull/13639/commits/1706a3fa8a3a6a4e9abe47892321f5971b050da2

File: WooCommerce/Classes/ViewRelated/Custom Fields/CustomFieldsDetails.swift (hunk 3 above)
Comment (hafizrahman) [reply WooCommerce/Classes/ViewRelated/Custom Fields/CustomFieldsDetails.swift]: Updated as part of https://github.com/woocommerce/woocommerce-ios/pull/13639/commits/1706a3fa8a3a6a4e9abe47892321f5971b050da2

File: WooCommerce/Classes/ViewRelated/Custom Fields/CustomFieldsDetails.swift (hunk 4 above)
Comment (hafizrahman) [reply WooCommerce/Classes/ViewRelated/Custom Fields/CustomFieldsDetails.swift]: Updated as part of https://github.com/woocommerce/woocommerce-ios/pull/13639/commits/1706a3fa8a3a6a4e9abe47892321f5971b050da2

**Review by itsmeichigo — APPROVED** [review]
Thanks for the updates - LGTM :shipit:
